File: 1720-crawler-log-folder/crawler-log-folder.ts
function minOperations(logs: string[]): number {
    let currentDepth = 0;

    const getLogType = (log: string): 'child' | 'parent' | 'current' => {
        switch (log) {
            case '../':
                return 'parent';
            case './':
                return 'current';
            default:
                return 'child';
        }
    };

    for (const log of logs) {
        switch (getLogType(log)) {
            case 'child':
                currentDepth++;
                break;
            case 'parent':
                if (currentDepth > 0) currentDepth--;
                break;
            case 'current':
                break;
        }
    }

    return currentDepth;
}

File: 1720-crawler-log-folder/crawler-log-folder.rs
impl Solution {
    pub fn min_operations(logs: Vec<String>) -> i32 {
        let mut current_depth = 0;

        for log in logs {
            match log.as_str() {
                "../" => {
                    if current_depth > 0 {
                        current_depth -= 1
                    }
                }
                "./" => {}
                _ => current_depth += 1,
            }
        }

        current_depth
    }
}